Update translation guideline text

JoKeyser opened the pull request:
Hi, I'd like to offer some fixes for the translation guideline.
My initial commit only contains "uncontroversial" fixes of spelling and some grammar.

If you want, I can also offer more opinionated edits:
1. Convert the text more to Markdown (currently it contains HTML tags and unnecessary Markdown-interpreted syntax, like double-spaces).
2. Try and simplify/fix the text content itself (e.g. it's not *quite* correct that you need a new Weblate account; Weblate offers alternative sign-in options).

JoKeyser committed:
Fix guideline text: Mainly spelling, some grammar.

boutinb commented:
Hi @JoKeyser Thank you really much for this. If you'd like to correct the markdown and simplify the content, gern!

JoKeyser committed:
Syntax: Use 1 line per sentence, obviate <br> tags

JoKeyser committed:
Avoid rescaling aspect of screenshots.

JoKeyser committed:
Update and restructure text contents:

- More (suitable) links.
- More onboarding / beginner-friendly explanations (hopefully).
- Keep all screenshots.
- Add named sub-sections and paragraphs.

JoKeyser committed:
Fix order of screenshots and some polishing.

JoKeyser committed:
Some last polishing, ready for review.

JoKeyser commented:
@boutinb , I restructured, corrected, and simplified the text.
I also added some details and my own opinion/experience that may be useful for beginners to Weblate, as this will serve as the potentially "first contact" for people who click the new link within JASP.
I tried not to change or delete the intent of the previous version, but it turns out I changed basically every line to some degree.

Please let me know if I should add or change something.

In case you want to merge, please squash the commits beforehand.

For convencience, here the link to my current version: https://github.com/JoKeyser/jasp-desktop/blob/fix-translation-guideline-text/Docs/development/jasp-guideline-translators.md

JoKeyser committed:
Ok one last polish.

AlexanderLyNL committed:
Added some additional information

AlexanderLyNL reviewed (approved):
Looks really great Johannes! Thanks for all your work. I added some additional comments to this PR, which you can find on your own repository I reckon.

AlexanderLyNL committed:
Minor changes

JoKeyser commented:
@AlexanderLyNL thank you for the fast and thorough review, I think all your improvements are great.
I think this is ready to be (squashed and) merged!

AlexanderLyNL commented:
Johannes I've updated the page on JASP https://jasp-stats.org/translation-guideline/ accordingly. Could you please send me a picture and a short description so I can add those to a "About the authors" section at the end of the page? See for instance, at the bottom of 

https://jasp-stats.org/2021/04/30/open-your-bachelor-thesis-using-jasp-and-introducing-open-data-analyses-while-teaching-using-jasp-two-tutorial-videos/

JoKeyser commented:
@AlexanderLyNL , thank you, also the page looks great!

Thank for mentioning me on that page; let me point out that the initial text was by @FransMeerhoff , I feel he should be linked as well.
Here a sentence to provide context: "Johannes is a postdoc researcher at the Psychology and and Sports Science department of the Justus Liebig University, Giessen, Germany."
Below I just upload a picture, you can change it to make it fit (and once you uploaded it, I will delete this from the comment):
*deleted*

AlexanderLyNL commented:
@JoKeyser could you please provide me with a square photo? Eventually it gets resized to 100 x 100, and I don't want to stretch your photo in a way that you'd be unhappy about.

JoKeyser commented:
@AlexanderLyNL thanks for the heads-up about the target size; I made a new one:
![jo100px](https://user-images.githubusercontent.com/8722846/133934911-6d7848ab-2ac4-40eb-bedd-a1a687d61360.jpg)
